pilot/pkg/model/service.go

```diff
@@ -791,10 +793,33 @@ type TrafficDistribution int
 const (
 	// TrafficDistributionAny allows any destination
 	TrafficDistributionAny TrafficDistribution = iota
-	// TrafficDistributionPreferClose prefers traffic in same region/zone/network if possible, with failover allowed.
-	TrafficDistributionPreferClose TrafficDistribution = iota
+	// TrafficDistributionPreferPreferSameZone prefers traffic in same zone, failing over to same region and then network.
+	TrafficDistributionPreferSameZone TrafficDistribution = iota
```

Comment:
Is there a reason you have all of these explicitly set to `iota`? You generally don't put a value here and let Go auto-increment it: https://go.dev/wiki/Iota.

Reply:
I do not like syntax